### PR TITLE
fix: Sanatize HTTP client spans & breadcrumbs

### PR DESCRIPTION
**Pull request by cleptric**
Closes https://github.com/getsentry/sentry-php/issues/1429

See https://develop.sentry.dev/sdk/data-handling/#spans & https://develop.sentry.dev/sdk/data-handling/#breadcrumbs.

refs: https://github.com/getsentry/team-webplatform-meta/issues/2

### CODE REVIEW

**Review by DAcodedBEAT — APPROVED** [review]

File: src/Tracing/GuzzleTracingMiddleware.php [hunk 1]
```diff
@@ -29,9 +30,20 @@ public static function trace(?HubInterface $hub = null): Closure
                     return $handler($request, $options);
                 }
 
+                $partialUri = Uri::fromParts([
+                    'scheme' => $request->getUri()->getScheme(),
+                    'host' => $request->getUri()->getHost(),
+                    'port' => $request->getUri()->getPort(),
+                    'path' => $request->getUri()->getPath(),
+                ]);
```
Comment (stayallive) [inline src/Tracing/GuzzleTracingMiddleware.php:38]: `l`: The only "worry" I have here is that a `MalformedUriException` is thrown for some reason. Although this might be totally unfounded since a URI was already there but maybe we want to try/catch anyway just to be 100% sure since not all PSR-7's might be created equal.

**Review by stayallive — APPROVED** [review]

File: src/Tracing/GuzzleTracingMiddleware.php (hunk 1 above)
Comment (cleptric) [reply src/Tracing/GuzzleTracingMiddleware.php:38]: Yeah, this might be a safer choice in the end. During my tests, the request normally fails with a cURL exception before this code runs.

Comment (stayallive) [reply src/Tracing/GuzzleTracingMiddleware.php:38]: Yeah that's why I figured this might be overkill, so it might not be 100% needed. I also wouldn't know what to fallback too, because dropping the breadcrumb also seems wrong... so maybe leave as is and see what edge cases fall out. Don't think this could every happen outside synthetic tests anyway.

**Comment by cleptric** [comment]
@retnek The changes here are unrelated, the transaction name changes were made in our Symfony SDK https://github.com/getsentry/sentry-symfony/pull/692.

Nonetheless, I'm sorry that you ran into this. We had to make these changes, as we got a lot of problems with people only seeing `<<unparameterized>>` transaction names on the performance page.
Please send an email to michi [at] sentry.io including your org name, so we can get you some transaction quota back.
